src/Raven.Server/Documents/Sharding/Handlers/ShardedOutgoingReplicationHandler.cs

```diff
@@ -78,7 +78,7 @@ protected override void Replicate()
             }
             catch (Exception e)
             {
-                batch?.BatchSent.TrySetException(e);
+                batch?.BatchSent?.TrySetException(e);
```

Comment:
Because this is disposed, we'll never actually send the error now, no?

Reply:
I think the shard that got the exception will set it, while the rest of the shard will be cancelled.. but yes, need to check that

Reply:
What do we see on the remote end?